app/Services/Allegro/AllegroService.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Services\Allegro;
+
+use App\Exceptions\InvalidBodyResponseException;
+use App\Models\Offer;
+use App\Services\AbstractOfferService;
+use Illuminate\Support\Facades\Http;
+use Spatie\Browsershot\Browsershot;
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\BrowserKit\HttpBrowser;
+use Symfony\Component\HttpClient\HttpClient;
+class AllegroService extends AbstractOfferService
+{
+    public array $classNames;
+    public string $body;
+    public string $apiProvider;
+    public string $apiKey;
+    public Offer $offer;
+    public function __construct(Offer $offer)
+    {
+        $this->classNames = config('theapp.allegro.data_provider.price_location.class_names');
+        $this->apiProvider = config('theapp.allegro.data_provider.api_provider');
+        $this->apiKey = config('theapp.allegro.data_provider.api_key');
+        $this->offer = $offer;
+    }
+
+    public function getOfferBody(): bool
+    {
+        $response = Http::timeout(121)->get($this->apiProvider, [
+            'api_key' => $this->apiKey,
+            'url' => $this->offer->url,
+        ]);
+
+        if(!$response->status() == 500) {
```

Comment:
To chyba nie jest poprawny warunek? Curl nie wyrzuca błędu dla statusu 400? może warto tutaj obsłużyć większą ilość błędów

Reply:
edytowałem te warunki, opisałem dokładniej o co mi chodziło tutaj: https://app.asana.com/0/1206212174113175/1206642564856432